packages/date-picker/test/validation.test.js

```diff
@@ -214,6 +214,24 @@ describe('validation', () => {
       datePicker._overlayContent._selectDate(new Date('2017-01-01')); // Invalid
     });
 
+    it('should reflect correct invalid value on value-changed eventListener when using isDateAvailable', (done) => {
```

Comment:
Some additional tests for the `max` attribute that should be adapted for the new function:
- https://github.com/vaadin/web-components/blob/877d1d27545bb3f1affe47c304c34001da5f9e6d/packages/date-picker/test/validation.test.js#L61-L67
- https://github.com/vaadin/web-components/blob/877d1d27545bb3f1affe47c304c34001da5f9e6d/packages/date-picker/test/validation.test.js#L132-L142